File: data/code_pairs/fortran/619.f90
! asyncwait.f90
program asyncwait
  implicit none
  integer, parameter :: N = 64
  real, allocatable :: a(:), b(:)
  integer :: i

  allocate (a(N))
  allocate (b(N))

  a(:) = 3.0
  b(:) = 0.0

  ! Assuming an operation to test here; 
  ! since none exists, we're just initializing arrays.
  !$acc parallel loop
  do i = 1, N
     b(i) = a(i) * 2.0
  end do

  !$acc wait async(N)

  ! Deallocate to clean up
  deallocate(a)
  deallocate(b)

end program asyncwait

File: data/code_pairs/cpp/619.cpp
// asyncwait.cpp
#include <vector>
#include <openacc.h>

int main() {
    const int N = 64;
    std::vector<float> a(N, 3.0f); // Initialize vector a with N elements of value 3.0
    std::vector<float> b(N, 0.0f); // Initialize vector b with N elements of value 0.0

    #pragma acc parallel loop
    for (int i = 0; i < N; ++i) {
        b[i] = a[i] * 2.0f;
    }

    acc_wait_async(N); // Wait on the asynchronous queue numbered N.

    return 0;
}